integration-tests/pkg/collector/introspection.go

```diff
@@ -23,3 +30,50 @@ func (k *K8sCollectorManager) IntrospectionQuery(endpoint string) ([]byte, error
 	body, err := io.ReadAll(resp.Body)
 	return body, err
 }
+
+func QueryConfig(t *testing.T, collectorIP string) []byte {
+	log.Info("Querying: /state/config")
+	body, err := IntrospectionQuery(collectorIP, "/state/config")
+	assert.NoError(t, err)
+	log.Info("Response: %q", body)
+	return body
+}
+
+func AssertExternalIps(t *testing.T, enable bool, collectorIP string) {
+	AssertRepeated(t, func() bool {
+		body := QueryConfig(t, collectorIP)
+		var response types.RuntimeConfig
+		err := json.Unmarshal(body, &response)
+		assert.NoError(t, err)
+
+		return response.Networking.ExternalIps.Enable == enable
+	})
+}
+
+func AssertNoRuntimeConfig(t *testing.T, collectorIP string) {
+	AssertRepeated(t, func() bool {
+		body := QueryConfig(t, collectorIP)
+		return strings.TrimSpace(string(body)) == "{}"
+	})
+}
+
+// TODO: This should be in its own package
+func AssertRepeated(t *testing.T, condition func() bool) {
+	tick := time.NewTicker(1 * time.Second)
+	timer := time.After(3 * time.Minute)
+
+	for {
+		select {
+		case <-tick.C:
+			if condition() {
+				// Condition has been met
+				return
+			}
+
+		case <-timer:
+			// TODO: This message should be passed in rather than hard coded here
+			t.Log("Timeout reached: Runtime configuration was not updated")
+			t.FailNow()
+		}
+	}
+}
```

Comment:
We probably want to move all assertions to a separate `pkg/assert` or similar.

Reply:
Done